pkg/diagnostics/http_monitoring.go

```diff
@@ -299,7 +300,15 @@ func (h *httpMetrics) HTTPMiddleware(next http.Handler) http.Handler {
 			endpointData, _ := r.Context().Value(endpoints.EndpointCtxKey{}).(*endpoints.EndpointCtxData)
 			method = endpointData.GetEndpointName()
 			if endpointData != nil && endpointData.Group != nil && endpointData.Group.MethodName != nil {
-				method = endpointData.Group.MethodName(r)
+				methodName := endpointData.Group.MethodName(r)
```

Comment:
This kind of "ad-hoc" things have bit us before in the past, where before #6919 the business logic for monitoring (logs, metrics, tracing) was a bit scattered all over the place, and it caused bugs and inconsistencies.

Instead of hardcoding method names in the `diagnostics` package, it'd be preferrable to add a new rule to EndpointGroup (https://github.com/dapr/dapr/blob/b9843e84a86b924255f9ab115c4d163cca9593b7/pkg/api/http/endpoints/endpointgroup.go#L21) which is a method that can be invoked (alternatively, to "Endpoint" if it's specific for an endpoint and not a group)

Then when the EndpointGroup/Endpoint [is defined](https://github.com/dapr/dapr/blob/b9843e84a86b924255f9ab115c4d163cca9593b7/pkg/api/http/actors.go#L73), the rule can be added there.

We do something similar for logs and other metrics today, for example for PubSub to append span attributes: https://github.com/dapr/dapr/blob/b9843e84a86b924255f9ab115c4d163cca9593b7/pkg/api/http/http.go#L264

Reply:
@ItalyPaleAle, I've updated the PR. Let me know if it makes more sense now. Thank you.

Reply:
@ItalyPaleAle ping?